src/_pytest/python.py

```diff
@@ -192,6 +194,13 @@ def pytest_pyfunc_call(pyfuncitem: "Function") -> Optional[object]:
     result = testfunction(**testargs)
     if hasattr(result, "__await__") or hasattr(result, "__aiter__"):
         async_warn_and_skip(pyfuncitem.nodeid)
+    elif result is not None:
+        warnings.warn(
+            PytestReturnNotNoneWarning(
+                f"Expected None, but the test returned {result!r}, which will be an error in a "
```

Comment:
The warning message should include the full node id of the test, otherwise it is hard for the user to find the offending test in a large suite.

Reply:
That makes sense, thanks!